admin_batch_update.php

```diff
@@ -1,24 +1,26 @@
-<?php
-	include "DBconnection.php";
-if (isset($_POST['updatedata'])){
-	 
-	$id = $_POST['update_id']; 
-		$batch_id = $_POST['batch_id'];
-		$batch_name = $_POST['batch_name'];
-		$batch_code = $_POST['batch_code'];
-		$course_id = $_POST['course_id'];
-		$assignment_id =$_POST['assignment_id'];
-
-   $query = "UPDATE student_batches SET batch_id='$batch_id', batch_name='$batch_name', batch_code='$batch_code', course_id='$course_id', course_id='$course_id', assignment_id='$assignment_id'  WHERE batch_id= $batch_id " ;
-	$query_run = mysqli_query($conn, $query);
-	if($query_run)
-	{	
-	echo "submitted";	
-   header('location:admin_batch.php') ;     	
-	}
-	else
-	{
-		echo "form not submitted";
-	}               
-}
-?>
\ No newline at end of file
+<?php
+	include "DBconnection.php";
+if (isset($_POST['updatedata'])){
+
+	$id = $_POST['update_id']; 
+	$batch_id = $_POST['batch_id'];
+	$batch_name = $_POST['batch_name'];
+	$batch_code = $_POST['batch_code'];
+	$course_id = $_POST['course_id'];
+	$assignment_id =$_POST['assignment_id'];
+
+	$query = "UPDATE student_batches SET batch_id=?, batch_name=?, batch_code=?, course_id=?, assignment_id=? WHERE batch_id=?";
+	$stmt = mysqli_prepare($conn, $query);
+	mysqli_stmt_bind_param($stmt, 'ssssss', $batch_id, $batch_name, $batch_code, $course_id, $assignment_id, $id);
+	$result = mysqli_stmt_execute($stmt);
+	if($result)
+	{   
+		echo "submitted";   
+		header('location:admin_batch.php') ;         
+	}
+	else
+	{
+		echo "form not submitted";
+	}               
+}
+?>

```

Comment:
The only possible improvement in this code is to remove the unnecessary check for $row > 0 since it always evaluates to true when the query results have data. Therefore, we can remove this check.

This code sets up an HTML table with a header row ( thead ) containing column headings for Batch ID, Batch Name, Batch Code, Course ID, Assignment ID, Edit and Delete columns. The body of the table ( tbody ) would typically contain rows of data with values corresponding to each of these columns.